### PR TITLE
REFACTOR-A-THON 3: introducing cats

### PR DESCRIPTION
**Pull request by gokyo**

### CODE REVIEW

File: src/main/scala/uk/gov/hmrc/fraudprevention/AntiFraudHeadersValidator.scala [hunk 1]
```diff
@@ -49,10 +47,13 @@ object AntiFraudHeadersValidator {
   }
 
   // To be called for each API incoming request
```
Comment (lburgos) [inline src/main/scala/uk/gov/hmrc/fraudprevention/AntiFraudHeadersValidator.scala]: As these are public methods that the library clients can use, should this information be in a scaladoc comment instead?

File: src/main/scala/uk/gov/hmrc/fraudprevention/headervalidators/HeaderValidator.scala [hunk 2]
```diff
@@ -16,24 +16,38 @@
...
+      case _ => Left(s"Multiple values for header $headerName")
+    }
   }
 
-  def headerName: String
+  protected def validateHeaderValue(headerValue: String): Either[String, Unit]
 
-  def isValidHeader(request: RequestHeader): Boolean
+  final def validate(request: RequestHeader): HeadersValidation = {
+    // Might be better to return Either[String, Unit] instead of ValidatedNel[String, Unit]
```
Comment (lburgos) [inline src/main/scala/uk/gov/hmrc/fraudprevention/headervalidators/HeaderValidator.scala]: Does this still apply? I think cats is already hidden from normal usage of the library through the AntiFraudHeadersValidator

File: src/main/scala/uk/gov/hmrc/fraudprevention/AntiFraudHeadersValidator.scala (hunk 1 above)
Comment (gokyo) [reply src/main/scala/uk/gov/hmrc/fraudprevention/AntiFraudHeadersValidator.scala]: ok

File: src/main/scala/uk/gov/hmrc/fraudprevention/headervalidators/HeaderValidator.scala (hunk 2 above)
Comment (gokyo) [reply src/main/scala/uk/gov/hmrc/fraudprevention/headervalidators/HeaderValidator.scala]: ok, I removed it

**Pull request merged**